torch/export/exported_program.py

```diff
@@ -307,6 +294,30 @@ def _decompose_and_get_gm_with_new_signature_constants(
     )
     from torch.fx.experimental.symbolic_shapes import ShapeEnv
 
+    # Note [Seperating decomp_table into CIA decomps and non-CIA decomps]
+    # At this point, we have a decomp_table that can contain None entries
+    # for ops user want to preserve in the aten IR.
+    # We need to separate the op into two categories:
+    # 1. CIA op: These are the ops that we want to override
+    #    CompositeImplicitAutograd decomp for. For them, we need to use _override_composite_implicit_decomp
+    #    context manager to plumb it through AOTDispatcher
+    # 2. Non-CIA op: These ops are only relevant after AOTDIspatcher runs, so just
+    #    checking if they are statically functional is enough.
+
+    preserve_cia_ops = []
+    for op in list(decomp_table.keys()):
```

Comment:
hmm. This doesn't seem like it will satisfy the default behavior we want, of "ops not present in the decomp table should never decompose"?

In particular, you are only adding ops with `None` decomps into the `preserve_cia_ops` list. Aren't you going to need to do something instead like this?

(1) loop through every CIA op registered to the dispatcher (e.g. with `torch._C._dispatch_print_registrations_for_dispatch_key("CompositeImplicitAutograd")`)

(2) for each CIA op, either:

(2a) preserve it if it is not in the decomp_table

(2b) if it is in the decomp_table, replace the existing CIA decomp with the user's